src/main/java/org/gradoop/benchmarks/tpgm/DiffBenchmark.java

```diff
@@ -0,0 +1,160 @@
+
+/**
+ * A dedicated program for parametrized TPGM graph evolution (diff) benchmark. The temporal
+ * predicate function is fixed to AS OF for this benchmark. The two query timestamps are specified
+ * during program arguments.
+ */
+public class DiffBenchmark extends BaseTpgmBenchmark {
+  /**
+   * Option to declare verification
+   */
+  private static final String OPTION_VERIFICATION = "v";
+  /**
+   * Option to declare the first query timestamp
+   */
+  private static final String OPTION_QUERY_1 = "x";
+  /**
+   * Option to declare the second query timestamp
+   */
+  private static final String OPTION_QUERY_2 = "y";
+
+  /**
+   * Used verification flag
+   */
+  private static boolean VERIFICATION;
+  /**
+   * Used first query timestamp in milliseconds
+   */
+  private static Long QUERY_FROM_1;
+  /**
+   * Used second query timestamp in milliseconds
+   */
+  private static Long QUERY_FROM_2;
+
+  static {
+    OPTIONS.addRequiredOption(OPTION_QUERY_1, "queryts1", true, "Used first query timestamp [ms]");
+    OPTIONS.addRequiredOption(OPTION_QUERY_2, "queryts2", true, "Used second query timestamp [ms]");
+    OPTIONS.addOption(OPTION_VERIFICATION, "verification", false, "Verify Snapshot with join.");
+  }
+
+  /**
+   * Main program to run the benchmark. Arguments are the available options.
+   * Example: {@code /path/to/flink run -c org.gradoop.benchmark.tpgm.SnapshotBenchmark
+   * path/to/gradoop-examples.jar -i hdfs:///graph -o hdfs:///output -c results.csv
+   * -x 1287000000000 -y 1230000000000}
+   *
+   * @param args program arguments
+   * @throws Exception in case of error
+   */
+  public static void main(String[] args) throws Exception {
+    CommandLine cmd = parseArguments(args, DiffBenchmark.class.getName());
+
+    if (cmd == null) {
+      return;
+    }
+
+    // read cmd arguments
+    readBaseCMDArguments(cmd);
+    readCMDArguments(cmd);
+
+    // create gradoop config
+    ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+    TemporalGradoopConfig conf = TemporalGradoopConfig.createConfig(env);
+
+    // read graph
+    TemporalDataSource source = new TemporalCSVDataSource(INPUT_PATH, conf);
+    TemporalGraph graph = source.getTemporalGraph();
+
+    // get the diff
+    TemporalGraph diff = graph.diff(new AsOf(QUERY_FROM_1), new AsOf(QUERY_FROM_2));
+
+    // apply optional verification
+    if (VERIFICATION) {
+      diff = diff.verify();
+    }
+
+    // write graph
+    writeOrCountGraph(diff, conf);
+
+    // execute and write job statistics
+    env.execute(DiffBenchmark.class.getSimpleName() + " - P: " + env.getParallelism());
+    writeCSV(env);
+  }
+
+  /**
+   * Reads the given arguments from command line
+   *
+   * @param cmd command line
+   */
+  private static void readCMDArguments(CommandLine cmd) {
+    QUERY_FROM_1 = Long.valueOf(cmd.getOptionValue(OPTION_QUERY_1));
+    QUERY_FROM_2 = Long.valueOf(cmd.getOptionValue(OPTION_QUERY_2));
```

Comment:
I think you can store these parameters as `long` to implicitly avoid `null`, you can then use `Long.parseLong` instead.